cmd/kogito/command/install/kogito_operator.go

```diff
@@ -0,0 +1,190 @@
+)
+
+var (
+	operatorImageName = fmt.Sprintf("%s:%s", defaultOperatorImageName, version.Version)
+)
+
+type installKogitoOperatorFlags struct {
+	Namespace string
+	Image     string
+}
+
+type installKogitoOperatorCommand struct {
+	context.CommandContext
+	flags   installKogitoOperatorFlags
+	command *cobra.Command
+	Parent  *cobra.Command
+}
+
+func newInstallKogitoOperatorCommand(ctx *context.CommandContext, parent *cobra.Command) context.KogitoCommand {
+	command := installKogitoOperatorCommand{
+		CommandContext: *ctx,
+		Parent:         parent,
+	}
+
+	command.RegisterHook()
+	command.InitHook()
+
+	return &command
+}
+
+func (i *installKogitoOperatorCommand) Command() *cobra.Command {
+	return i.command
+}
+
+func (i *installKogitoOperatorCommand) RegisterHook() {
+	i.command = &cobra.Command{
+		Use:     "operator [flags]",
+		Short:   "Installs the Kogito Operator into the OpenShift/Kubernetes cluster",
+		Example: "install operator -p my-project",
+		Long: `Installs the Kogito Operator via OLM if available by subscripting to the Kogito Operator.
+					  If the operator is not available at OLM, will use the traditional deployment via Deployment resource.`,
+		RunE:    i.Exec,
+		PreRun:  i.CommonPreRun,
+		PostRun: i.CommonPostRun,
+		Args: func(cmd *cobra.Command, args []string) error {
+			return nil
+		},
+	}
+}
+
+func (i *installKogitoOperatorCommand) InitHook() {
+	i.flags = installKogitoOperatorFlags{}
+	i.Parent.AddCommand(i.command)
+	i.command.Flags().StringVarP(&i.flags.Namespace, "project", "p", "", "The project name where the operator will be deployed")
+	i.command.Flags().StringVarP(&i.flags.Image, "image", "i", operatorImageName, "The operator image")
+}
+
+func (i *installKogitoOperatorCommand) Exec(cmd *cobra.Command, args []string) error {
+	log := context.GetDefaultLogger()
+	config := context.ReadConfig()
+
+	var err error
+	if i.flags.Namespace, err = shared.EnsureProject(i.Client, config, i.flags.Namespace); err != nil {
+		return err
+	}
+
+	if exists, err := shared.CheckKogitoOperatorExists(i.Client, i.flags.Namespace); err != nil {
+		return err
+	} else if exists {
+		log.Infof("Kogito Operator is already deployed in the namespace '%s', skipping ", i.flags.Namespace)
+		return nil
+	}
+
+	//TODO: check if we have operator available at the hub, warn the user and exit
+
+	log.Infof("Kogito Operator not found in the namespace '%s', trying to deploy it", i.flags.Namespace)
+
+	if err := i.installOperatorWithYamlFiles(); err != nil {
+		return fmt.Errorf("Error while deploying Kogito Operator via template yaml files: %s ", err)
+	}
+
+	log.Infof("Kogito Operator successfully deployed in '%s' namespace", i.flags.Namespace)
+
+	return nil
+}
+
+func (i *installKogitoOperatorCommand) installOperatorWithYamlFiles() error {
+	box := packr.New("deploy", boxDeployPath)
+
+	if err := i.decodeAndCreateKubeObject(box, fileKogitoAppCRDYaml, &apiextensionsv1beta1.CustomResourceDefinition{}, nil); err != nil {
```

Comment:
Will the exception be thrown in case CRD is already available on cluster?

Reply:
Nops, because we call the `CreateIfNotExists` function from our API wrapper.

Reply:
ok, thanks